crates/matrix-sdk/src/notification_settings.rs

```diff
@@ -0,0 +1,1156 @@
+//! High-level push notification settings API
+
+use ruma::{
+    api::client::push::{delete_pushrule, set_pushrule, set_pushrule_enabled, RuleScope},
+    push::{
+        Action, NewConditionalPushRule, NewPushRule, NewSimplePushRule, PredefinedContentRuleId,
+        PredefinedOverrideRuleId, PredefinedUnderrideRuleId, PushCondition, RuleKind, Ruleset,
+        Tweak,
+    },
+    RoomId,
+};
+
+use crate::{error::NotificationSettingsError, Client, Result};
+
+/// Enum representing the push notification modes for a room
+#[derive(Debug, Clone, PartialEq)]
+pub enum RoomNotificationMode {
+    /// All messages
+    AllMessages,
+    /// Mentions and keywords only
+    MentionsAndKeywordsOnly,
+    /// Mute
+    Mute,
+}
+
+/// A high-level API to manage the client owner's push notification settings.
+#[derive(Debug, Clone)]
+pub struct NotificationSettings {
+    /// The underlying HTTP client.
+    client: Client,
+}
+
+impl NotificationSettings {
+    pub(crate) fn new(client: Client) -> Self {
+        Self { client }
+    }
+
+    /// Sets a notification mode for a given room
+    pub async fn set_room_notification_mode(
+        &self,
+        room_id: &String,
```

Comment:
```suggestion
        room_id: &RoomId,
```

would be better, as we ensure the `room_id` is syntactically correct.

Reply:
Yes, it's definitely better. I've updated the code so that all functions now take a `RoomId` instead of a `String` as an input parameter.